force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -139,15 +196,21 @@ public with sharing class FlowControlLoopsCollections {
      */
     public static String returnHelloWorld() {
         // Initialize the result String
+        StringBuilder result = new StringBuilder();
 
         // Use a for loop to append 'Hello World!;' to the result string three times
         /* 
-         for (){
+        for (){
             
-         }
+        }
         */
-
-        return null; // Replace null with the variable you used to store the result
+        for (int i = 0; i < 3; i++) {
+            result.append("Hello World!");
+        if (i < 2) {
+            result.append(''; '');
+        }
+        }     
+        return result.toString(); // Replace null with the variable you used to store the result
```

Comment:
Apex does not have a built-in StringBuilder class. If you wrote your own, feel free to add it to the assignment.  But  take a look at the buildtin Apex string methods.  You can solve this with out the nested loop

https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_string.htm